### DISCUSSION TITLE
Simple script using wrong timezone on Mac OSX Sonoma 14.3 in Australia/Melbourne

### DISCUSSION THREAD

**Discussion by jmatthewpryor** [johnlindquist/kit#1430] [Errors] [answered]
Hi
I am in 'Australia/Melbourne' and ever since DST started, this simple script returns the time for UTC+10 rather than UTC+11
```
// Name: TimeToFive
// Shortcut: cmd shift 5
// Author: J. Matthew Pryor
// Twitter: @jmatthewpryor

import "@johnlindquist/kit"

    const now = new Date();
    let minutes = now.getMinutes();
    let hours = now.getHours();

    // Round the minutes to the nearest 5
    minutes = Math.round(minutes / 5) * 5;

    // If rounding up moves us to the next hour, adjust the hour
    if (minutes === 60) {
      minutes = 0;
      hours += 1;
    }

    // Format the time as HH:MM
    const formattedHours = hours.toString().padStart(2, "0");
    const formattedMinutes = minutes.toString().padStart(2, "0");
    const timeToFive = `${formattedHours}:${formattedMinutes}`;

    await copy(timeToFive);
    await setSelectedText(timeToFive);
  
```
I assume it is some node/OSX level problem, and there were also issues with Slack and a few other tools for a while - I am less clear how to fix it for ScriptKit

**Comment by jmatthewpryor** [comment]
Any tips for how to track this down @johnlindquist ?

**Reply by johnlindquist** [reply]
@jmatthewpryor I asked ChatGPT and got this and it seems to work... I haven't had to deal with dates/timezones in a while

```js
import "@johnlindquist/kit";

const now = new Date();
const options = { timeZone: 'Australia/Melbourne', hour: '2-digit', minute: '2-digit' };
const timeString = new Intl.DateTimeFormat('en-AU', options).format(now);

let [hours, minutes] = timeString.split(':').map(num => parseInt(num, 10));

// Round the minutes to the nearest 5
minutes = Math.round(minutes / 5) * 5;

// If rounding up moves us to the next hour, adjust the hour
if (minutes === 60) {
  minutes = 0;
  hours += 1;
}

// Ensure hours wrap around if they exceed 23
if (hours >= 24) {
  hours -= 24;
}

// Format the time as HH:MM
const formattedHours = hours.toString().padStart(2, "0");
const formattedMinutes = minutes.toString().padStart(2, "0");
const timeToFive = `${formattedHours}:${formattedMinutes}`;

await copy(timeToFive);
await setSelectedText(timeToFive);
```

**Reply by jmatthewpryor** [reply]
thanks @johnlindquist - appreciate the fix. The issue I was trying to track down seems to be related to Node and the OS - i.e. many tools that use Node on the Mac had this issue for a while, but each of them gradually 'corrected' - I have a local nextjs based server that I used for certain integrations and it reports the correct TZ.

I guess one other question would be - how do I know what version of Node Kit is using?

**Reply by johnlindquist** [reply, marked as answer]
@jmatthewpryor From your terminal:


~/.knode/bin/node --verison

On my phone, but pretty sure it's 18.18.2

**Reply by jmatthewpryor** [reply]
thanks @johnlindquist FWIW it seems to have been a node version issue since now the original code returns the correct result (i.e. DST adjusted time) - thanks
